File: src/main/java/com/felix/bms/entity/Comment.java
package com.felix.bms.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "comments")
@Getter @Setter
public class Comment implements Likeable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(columnDefinition = "TEXT",  nullable = false)
    private String content;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "author_id",nullable = false)
    private User author;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "post_id", nullable = false)
    private BlogPost post;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_comment_id")
    private Comment parentComment; // For replies

    @OneToMany(mappedBy = "parentComment", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Comment> replies = new ArrayList<>();

    @OneToMany(mappedBy = "comment", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Like> likes = new ArrayList<>(); // Add this for comment likes

    private LocalDateTime createdAt=LocalDateTime.now();
}

File: types/comment.ts
import type { User } from './user'

export interface Comment {
  id: string
  articleId: string
  author: Pick<User, 'id' | 'username' | 'name' | 'avatar'>
  content: string
  likes: number
  isLiked: boolean
  replies: CommentReply[]
  createdAt: string
}

export interface CommentReply {
  id: string
  commentId: string
  author: Pick<User, 'id' | 'username' | 'name' | 'avatar'>
  content: string
  likes: number
  isLiked: boolean
  createdAt: string
}
